### PR TITLE
Rebase 1.4

### PR DESCRIPTION
**Pull request by kaixuanliu**
rebase to [[text-embeddings-inference](https://github.com/huggingface/text-embeddings-inference/tree/main)] v1.4.0 tag

### CODE REVIEW

File: .github/workflows/build_all.yaml
Comment (regisss) [inline .github/workflows/build_all.yaml:1]: Let's remove all the workflows except `trufflehog.yml` and `push_docker_image.yml`. I had to enable Github Actions in this repo for using `push_docker_image.yml` but other workflows won't work in this fork so better to remove them.

**Review by regisss — APPROVED** [review]
LGTM!

**Comment by pi314ever** [comment]
I think `Cargo.lock` also needs to be updated for this to be complete.

**Comment by kaixuanliu** [comment]
> I think `Cargo.lock` also needs to be updated for this to be complete.

@pi314ever Oh, yes, you can submit a PR to fix it, or add it along with another PR.

**Pull request merged**